src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,133 @@
+package parser;
+
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+import java.util.Optional;
+import task.Deadline;
+import task.EmptyDescriptionException;
+import task.Event;
+import task.Task;
+import task.TaskTypeEnum;
+import task.ToDo;
+
+public class Parser implements IParser {
+    private Scanner sc;
+    private String message;
+    private Command currentCommand;
+    public Parser(Scanner sc) {
+        this.sc = sc;
+    }
+    @Override
+    public String getMessage() {
+        return message;
+    }
+    private boolean isList() {
+        return (message.split(" ")[0].toLowerCase().equals("list"));
+    }
+    private boolean isExit() {
+        return (message.split(" ")[0].toLowerCase().equals("bye"));
+    }
+    private boolean isMark(){
+        return (message.split(" ")[0].toLowerCase().equals("mark"));
+    }
+    private boolean isUnmark(){
+        return (message.split(" ")[0].toLowerCase().equals("unmark"));
+    }
+    private TaskTypeEnum typeOfTask() throws InvalidCommandException {
+        String task = message.split(" ")[0].toUpperCase();
+        Optional<TaskTypeEnum> answer = Arrays.stream(TaskTypeEnum.values())
+                                        .filter(x -> x.toString().equals(task))
+                                        .findFirst();
+        if (answer.isPresent()) {
+            return answer.get();
+        }
+        else {
+            throw new InvalidCommandException("Invalid format!!!\nKeyword must be in the form of <Keyword> <Args>", new IllegalArgumentException());
+        }
```

Comment:
this line seems too long, maybe you can saperate it in to two lines.

Reply:
Yeah, agreed. Im not sure if ill be breaking any coding standard if i were to break it into 2 lines. nothing on the website about doing this. Was thinking about having this statement span over 3 lines for readbility. Thanks